Utils/azuretools-core/src/com/microsoft/azure/toolkit/lib/common/operation/AzureOperationsContext.java

```diff
@@ -58,7 +59,9 @@ public static Object execute(final AzureOperationRef operation, final OperationP
         try {
             return proceedable.proceed();
         } finally {
-            AzureOperationsContext.pop();
+            if (CollectionUtils.isNotEmpty(operations.get())) {
```

Comment:
this should be impossible, when will this be empty?

Reply:
for #1797882, we found some actions will execute in same thread, which will clear the operations stack, so check whether operations is empty here